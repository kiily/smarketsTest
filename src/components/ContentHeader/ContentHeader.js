import React from 'react';
import PropTypes from 'prop-types';
import './ContentHeader.scss';

const contentHeader = (props) => {
  const currentSportText = `${props.currentSport.charAt(0).toUpperCase()}${props.currentSport.slice(1)}`;
  return (
    <div className="ContentHeader-content">
      <h1 className="title">{currentSportText} odds</h1>
      <p className="description">
        <span>
            Smarkets betting exchange allows you to bet with the best <span>{currentSportText}</span>
            odds - thanks to our small margins and industry-low 2% commission - on all tournaments and competitions
          {props.currentSport === 'football' && <span> including Premier League, La Liga, Europa League and Champions League.</span>}
        </span>
      </p>
    </div>
  );
};

contentHeader.propTypes = {
  currentSport: PropTypes.string
};

export default contentHeader;
