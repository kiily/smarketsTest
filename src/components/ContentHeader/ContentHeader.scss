@import '../../styles/_variables.scss';

.ContentHeader-content {
    color: $white;
    padding: 0 1.25rem;
    width: 60%;
    .title {
        display: flex;
        align-items: center;
        .favorite-button {
            background: transparent;
            border: none;
            color: $grey;
            outline: none;
            margin-left: 1rem;
            padding: 0.25rem 0;
            
            &.favorite {

            }
        }
    }
}