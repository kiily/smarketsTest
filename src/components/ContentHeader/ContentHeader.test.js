import React from 'react';
import { create, act } from 'react-test-renderer';
import ContentHeader from './ContentHeader';

describe('ContentHeader component', () => {
  let component;
  let rootInstance;
  beforeEach(() => {
    component = create(<ContentHeader currentSport="football" />);
    rootInstance = component.root;
  });

  it('should match the snapshot when rendered with props', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should toggle the favorite class after the favorite button is clicked', () => {
    const favoriteButton = rootInstance.findByType('button');
    act(() => {
      favoriteButton.props.onClick();
    });
    expect(favoriteButton.props.className).toBe('button-custom favorite');
    expect(component.toJSON()).toMatchSnapshot();
  });
});
