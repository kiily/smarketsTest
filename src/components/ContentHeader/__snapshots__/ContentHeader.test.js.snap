// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHeader component should match the snapshot when rendered with props 1`] = `
<div
  className="ContentHeader-content"
>
  <h1
    className="title"
  >
    Football
     odds
    <button
      className="button-custom "
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </h1>
  <p
    className="description"
  >
    <span>
      Smarkets betting exchange allows you to bet with the best 
      <span>
        Football
      </span>
      odds - thanks to our small margins and industry-low 2% commission - on all tournaments and competitions
      <span>
         including Premier League, La Liga, Europa League and Champions League.
      </span>
    </span>
  </p>
</div>
`;

exports[`ContentHeader component should toggle the favorite class after the favorite button is clicked 1`] = `
<div
  className="ContentHeader-content"
>
  <h1
    className="title"
  >
    Football
     odds
    <button
      className="button-custom favorite"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-star fa-w-18 "
        data-icon="star"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </h1>
  <p
    className="description"
  >
    <span>
      Smarkets betting exchange allows you to bet with the best 
      <span>
        Football
      </span>
      odds - thanks to our small margins and industry-low 2% commission - on all tournaments and competitions
      <span>
         including Premier League, La Liga, Europa League and Champions League.
      </span>
    </span>
  </p>
</div>
`;
